backend/src/case_file/case_file.service.ts

```diff
@@ -906,38 +815,96 @@ export class CaseFileService {
     return `This action removes a #${id} caseFile`;
   }
 
-  createNote = async (
-    input: CreateSupplementalNoteInput
-  ): Promise<CaseFile> => {
+  createNote = async (model: CreateSupplementalNoteInput): Promise<CaseFile> => {
     let caseFileId = "";
 
-    const { leadIdentifier, note, createUserId, actor } = input;
+    const { leadIdentifier, note, createUserId, actor } = model;
     const caseFile = await this.findOneByLeadId(leadIdentifier);
 
     if (caseFile.caseIdentifier) {
       caseFileId = caseFile.caseIdentifier;
     } else {
-      const caseInput: CreateCaseInput = { ...input };
+      const caseInput: CreateCaseInput = { ...model };
       caseFileId = await this.createCase(caseInput);
     }
 
     return await this._upsertNote(caseFileId, note, actor, createUserId);
   };
 
-  updateNote = async (
-    input: UpdateSupplementalNoteInput
-  ): Promise<CaseFile> => {
-    const { caseIdentifier: caseFileId, actor, note, updateUserId } = input;
+  updateNote = async (model: UpdateSupplementalNoteInput): Promise<CaseFile> => {
+    const { caseIdentifier: caseFileId, actor, note, updateUserId } = model;
 
     return await this._upsertNote(caseFileId, note, actor, updateUserId);
   };
 
-  private _upsertNote = async (
-    caseId: string,
-    note: string,
-    actor: string,
-    userId: string
-  ): Promise<CaseFile> => {
+  deleteNote = async (model: DeleteSupplementalNoteInput): Promise<CaseFile> => {
+    const _getNoteActionXref = async (): Promise<string> => {
```

Comment:
I think it would be faster if you're passing the `action_guid` in DeleteSupplementalNoteInput so that you don't have to query `action_type_action_xref` and `action` table again just to find the id

Reply:
Other than that, the rest looks good to me

Reply:
@Scarlett-Truong are we doing this anywhere else?